test/nodes/test_shaper.py

```diff
@@ -0,0 +1,563 @@
+from haystack import Pipeline, Document
+
+
+def test_basic_function_invocation(tmp_path):
+    with open(tmp_path / "tmp_config.yml", "w") as tmp_file:
+        tmp_file.write(
+            f"""
+            version: ignore
+            components:
+            - name: shaper
+              params:
+                inputs:
+                    query:
+                      func: expand
+                      output: questions
+                      params:
+                        expand_target: query
+                        size:
+                          func: len
+                          params:
+                            - documents
+                    documents:
+                      func: concat_docs
+                      output: documents
+                      params:
+                        docs: documents
+                        delimiter: " "
+              type: Shaper
+            pipelines:
+              - name: query
+                nodes:
+                  - name: shaper
+                    inputs:
+                      - Query
+        """
+        )
+    pipeline = Pipeline.load_from_yaml(path=tmp_path / "tmp_config.yml")
+    result = pipeline.run(
+        query="What can you tell me about Berlin?",
+        documents=[Document("Berlin is an amazing city."), Document("I love Berlin.")],
+    )
+    assert result
+    assert isinstance(result["query"], str)
+    questions = result["meta"]["invocation_context"]["questions"]
+
+    # questions has been expanded to a list of strings of size 2 (because Documents has 2 elements)
+    assert isinstance(questions, list) and len(questions) == 2 and questions[0] == result["query"]
```

Comment:
Could you also add asserts checking the output of `documents`? We should also check the behavior for `concat_docs` is working as expected.